cpp/src/binaryop/compiled/operation.cuh

```diff
@@ -216,6 +216,29 @@ struct Pow {
   }
 };
 
+struct IntPow {
+  template <
+    typename TypeLhs,
+    typename TypeRhs,
+    std::enable_if_t<(std::is_integral_v<TypeLhs> and std::is_integral_v<TypeRhs>)>* = nullptr>
+  __device__ inline auto operator()(TypeLhs x, TypeRhs y) -> int
+  {
+    cudf_assert(y >= 0 && "integer exponentiation with negative exponent is not possible.");
```

Comment:
Proposing to remove the `cudf_assert` because we can't really exit from device code.
```suggestion
    if (y < 0) {
      // Integer exponentiation with negative exponent is not possible.
      return 0;
    }
```

Reply:
Great. Done!